### PR TITLE
feat (core): Add api for usb events interfacing

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
Fixes https://app.clickup.com/t/9002019994/CHI-4722

### CODE REVIEW

File: common/interfaces/desktop_app_interface/usb_event.c [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ * @brief   USB Event APIs.
+ *          Describes all the logic for interfacing with USB Events.
+ * @copyright Copyright (c) 2023 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/"
+ *target=_blank>https://mitcc.org/</a>
+ *
+ ******************************************************************************
+ * @attention
+ *
+ * (c) Copyright 2022 by HODL TECH PTE LTD
```
Comment (kartikCypherock) [inline common/interfaces/desktop_app_interface/usb_event.c:13]:
```suggestion
 * (c) Copyright 2023 by HODL TECH PTE LTD
```

File: common/interfaces/desktop_app_interface/usb_event.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#include "communication.h"
+
+/*****************************************************************************
+ * MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * TYPEDEFS
+ *****************************************************************************/
+typedef struct usb_event {
```
Comment (kartikCypherock) [inline common/interfaces/desktop_app_interface/usb_event.h:28]: Soft suggestion to start names of the members with `usb_`

Maybe:
```
typedef struct {
  uint8_t usb_event_flag;
  En_command_type_t usb_cmd_id;
  uint16_t usb_msg_size;
  uint8_t *p_usb_msg;
} usb_event_t;
```

Comment (ujjwal-cyph) [reply common/interfaces/desktop_app_interface/usb_event.h:28]: I will add `_t`. But

the member names without `usb_event_` suffix seem better. Example:

```
usb_event_t usb_event;

usb_event.usb_event_flag = true;
usb_event.flag = true;
```

File: common/interfaces/desktop_app_interface/usb_event.c (hunk 1 above)
Comment (ujjwal-cyph) [reply common/interfaces/desktop_app_interface/usb_event.c:13]: fixed

File: common/interfaces/desktop_app_interface/usb_event.h (hunk 2 above)
Comment (ujjwal-cyph) [reply common/interfaces/desktop_app_interface/usb_event.h]: changed

**Review by kartikCypherock — APPROVED** [review]

**Comment by rohanagarwal94** [comment]
Task linked: [CHI-4722 Implement USB event getter for firmware](https://app.clickup.com/t/9002019994/CHI-4722)

**Pull request merged**